### PR TITLE
Feature/trade status

### PR DESCRIPTION
**Pull request by martineehrlich**
## Description
A few sentences describing the overall goals of the pull request's commits.

## Related PRs
List related PRs if applicable

## Todos
- [x] Tests
- [x] Documentation
- [x] Link to Trello

### CODE REVIEW

**Comment by martineehrlich** [comment]
[Trading Status for Brokers](https://trello.com/c/phnPA11m/603-trading-status-for-brokers)

File: broker-cli/commands/wallet.js [hunk 1]
```diff
@@ -201,13 +202,83 @@ async function networkAddress (args, opts, logger) {
...
+  if (Big(balance).gt(0)) {
+    switch (status) {
+      case 'active':
+        return fixedBalance.green
+      case 'pending':
+        return fixedBalance.yellow
+      case 'inactive':
+        return fixedBalance.red
+    }
+  } else {
```
Comment (treygriffith) [inline broker-cli/commands/wallet.js]: you don't need an else if the if returns in every case

File: broker-cli/commands/wallet.js [hunk 2]
```diff
@@ -201,13 +202,83 @@ async function networkAddress (args, opts, logger) {
...
+ *
+ * @function
+ * @param {string} balance
+ * @param {string} status
+ * @return {string} balance formatted with size and color
+ */
+function formatBalance (balance, status) {
+  const fixedBalance = Big(balance).toFixed(16)
+  if (Big(balance).gt(0)) {
+    switch (status) {
```
Comment (treygriffith) [inline broker-cli/commands/wallet.js]: is there a default case here? What is it?

File: broker-cli/commands/wallet.js [hunk 3]
```diff
@@ -201,13 +202,83 @@ async function networkAddress (args, opts, logger) {
...
+
 module.exports = (program) => {
   program
     .command('wallet', 'Commands to handle a wallet instance')
-    .help('Available Commands: balance, new-deposit-address, commit-balance, network-address')
+    .help('Available Commands: balance, new-deposit-address, commit-balance, network-address, network-status')
     .argument('<command>', '', Object.values(SUPPORTED_COMMANDS), null, true)
     .argument('[sub-arguments...]')
     .option('--rpc-address', 'Location of the RPC server to use.', validations.isHost)
+    .option('--market [marketName]', 'Relevant market name', validations.isMarketName)
```
Comment (treygriffith) [inline broker-cli/commands/wallet.js]: don't you want to put this below network-status?

File: broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  if (!counterEngine) {
+    throw new Error(`No engine available for ${counterSymbol}`)
+  }
+
+  const [baseSymbolCapacities, counterSymbolCapacities] = await Promise.all([
+    getEngineTradingCapacities([baseSymbol, baseEngine]),
+    getEngineTradingCapacities([counterSymbol, counterEngine])
+  ])
+
+  console.log(baseSymbolCapacities, counterSymbolCapacities)
```
Comment (treygriffith) [inline broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js]: leftover log?

File: broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    activeReceiveCapacity: Big(activeChannelCapacities.remoteBalance).div(divideBy).toString(),
+    pendingSendCapacity: Big(pendingChannelCapacities.localBalance).div(divideBy).toString(),
+    pendingReceiveCapacity: Big(pendingChannelCapacities.remoteBalance).div(divideBy).toString(),
+    inactiveSendCapacity: Big(inactiveChannelCapacities.localBalance).div(divideBy).toString(),
+    inactiveReceiveCapacity: Big(inactiveChannelCapacities.remoteBalance).div(divideBy).toString()
+  }
+}
+
+/**
+ * Grabs the daemons lnd wallet balance
```
Comment (treygriffith) [inline broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js]: wrong doc

File: broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ * @function
+ * @param {GrpcUnaryMethod~request} request - request object
+ * @param {Object} request.params
+ * @param {Map} request.engines
+ * @param {Logger} request.logger
+ * @param {function} responses.GetTradingCapacitiesResponse
+ * @return {GetTradingCapacitiesResponse}
+ */
+async function getTradingCapacities ({ params, logger, engines }, { GetTradingCapacitiesResponse }) {
+  const [ baseSymbol, counterSymbol ] = params.market.split('/')
```
Comment (treygriffith) [inline broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js]: do we need to check that this is a valid market?

File: broker-daemon/proto/broker.proto [hunk 7]
```diff
@@ -895,4 +928,6 @@ service WalletService {
    * ```
    */
   rpc GetPaymentChannelNetworkAddress (GetPaymentChannelNetworkAddressRequest) returns (GetPaymentChannelNetworkAddressResponse);
+  rpc GetTradingCapacities (GetTradingCapacitiesRequest) returns (GetTradingCapacitiesResponse);
```
Comment (treygriffith) [inline broker-daemon/proto/broker.proto]: Add docs for this rpc, including javascript and shell examples

File: broker-daemon/proto/broker.proto [hunk 8]
```diff
@@ -725,6 +725,39 @@ message GetBalancesResponse {
...
+  string inactive_receive_capacity = 7;
+}
+
+/**
+ * All trading capacities for a Broker.
+ */
+message GetTradingCapacitiesResponse {
+  // Balances in all currencies available to this Broker.
+  Capacity base_symbol_capacities = 1;
+  Capacity counter_symbol_capacities = 2;
```
Comment (treygriffith) [inline broker-daemon/proto/broker.proto]: add comment

File: broker-daemon/proto/broker.proto [hunk 9]
```diff
@@ -725,6 +725,39 @@ message GetBalancesResponse {
...
+}
+
+message GetTradingCapacitiesRequest {
```
Comment (treygriffith) [inline broker-daemon/proto/broker.proto]: Add comments

File: broker-daemon/proto/broker.proto [hunk 10]
```diff
@@ -725,6 +725,39 @@ message GetBalancesResponse {
...
+  string market = 1;
```
Comment (treygriffith) [inline broker-daemon/proto/broker.proto]: add comments

File: broker-cli/commands/wallet.js (hunk 2 above)
Comment (martineehrlich) [reply broker-cli/commands/wallet.js]: No

File: broker-cli/commands/wallet.js (hunk 3 above)
Comment (martineehrlich) [reply broker-cli/commands/wallet.js]: The options only work if they are above the action.

File: broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js (hunk 6 above)
Comment (martineehrlich) [reply broker-daemon/broker-rpc/wallet-service/get-trading-capacities.js]: Is there another place that we do this?

File: broker-cli/commands/wallet.js (hunk 2 above)
Comment (treygriffith) [reply broker-cli/commands/wallet.js]: what's the behavior if status is not a value that you're expecting? Seems to me that you should handle it or throw an error or something

File: broker-cli/proto/broker.proto [hunk 11]
```diff
@@ -899,4 +937,78 @@ service WalletService {
    * ```
    */
   rpc GetPaymentChannelNetworkAddress (GetPaymentChannelNetworkAddressRequest) returns (GetPaymentChannelNetworkAddressResponse);
+  /**
+   * GetTradingCapacities retrieves the remote and local capacities from the base and counter engines active, pending and inactive channels.
+   *
+   activeSendCapacity: Big(activeChannelCapacities.localBalance).div(divideBy).toString(),
```
Comment (treygriffith) [inline broker-cli/proto/broker.proto]: I don't think you meant to include this

File: broker-cli/commands/wallet.js (hunk 3 above)
Comment (dannypaz) [reply broker-cli/commands/wallet.js]: @treygriffith because of the issues w/ sub-commands, we have to put it in both places unfortunately

**Pull request merged**